src/packaging/tags.py

```diff
@@ -124,20 +124,23 @@ def _normalize_string(string: str) -> str:
     return string.replace(".", "_").replace("-", "_").replace(" ", "_")
 
 
-def _abi3_applies(python_version: PythonVersion) -> bool:
+def _abi3_applies(python_version: PythonVersion, abis: List[str]) -> bool:
     """
     Determine if the Python version supports abi3.
 
-    PEP 384 was first implemented in Python 3.2.
+    PEP 384 was first implemented in Python 3.2. The `--disable-gil` builds
+    do not support abi3 and are indicated by a "t" (for "threading") in their
+    abi tags.
     """
-    return len(python_version) > 1 and tuple(python_version) >= (3, 2)
+    return (len(python_version) > 1 and tuple(python_version) >= (3, 2) and
+            (len(abis) == 0 or "t" not in abis[0]))
```

Comment:
This seems a bit brittle if you're only going to check for "t" somehow existing in the ABI tag (and anywhere in the tag on top of it).

Why not pass in a boolean for some argument named e.g. `threading` which automatically makes `abi3` not work since it's for a threaded/no-GIL build?

Reply:
I've changed the code to take in a `threading` argument, but it still needs to compute that somehow. I made the code that checks the abiflags a bit more conservative in that it only checks the ABI flags and not the whole ABI tag.

Reply:
Is `_get_config_var("Py_NOGIL", warn)` the canonical way to check if a threaded build is being used? If so, is there any reason why that can't be used to calculate the `threading` argument every time instead of parsing the ABI tags?

Reply:
That's what I was planning to do originally, but that only works for the current interpreter. Some of the public APIs (and may of the tests) allow you to compute tags for a different Python version or ABI than the one currently running.

Reply:
@brettcannon, does the above explanation make sense? To give a concrete example adapted from the tests:

The following call should not depend on current interpreter because we explicitly pass in the Python version, ABI, and platform. If we relied, on `_get_config_var("Py_NOGIL", warn)` than we might incorrectly exclude ABI3 tags. 

```python
tags.cpython_tags((3, 13), ["cp313"], ["macosx_13_0_x86_64"])
```

Reply:
> does the above explanation make sense?

Yep! But what I was getting at was in the places where you are calling into functions that need to know whether you're using a threaded build, could we just pass in the value by calculating `_get_config_var()` if it wasn't explicitly passed in earlier? Since there are only so many top-level functions, I don't think it would be bad to pass it in explicitly into e.g. `cpython_tags()`.

The reason I'm questioning this is the wheel tag is meant to be an opaque string, so I have tried to avoid any form of parsing them where possible. Granted, we control this code and it's probably fine for `cpython_tags()` since we control those tags implicitly compared to e.g. `generic_tags()`.

I'll give it a think and let you know no later than Friday.